### PR TITLE
Levels

### PR DESCRIPTION
**Pull request by axelmunch**
- Loading levels
- Displaying the loaded level

### CODE REVIEW

File: levels.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    return brick;
+}
+
+void load_level(const char *filename) {
+    // Reset level
+    level.theme = THEME_1;
+    level.offset = 0;
+    level.height = 0;
```
Comment (maelehrhard) [inline levels.c]: maybe you could this in a void method named reset_level() to remove the comment

File: arkanoid.c [hunk 2]
```diff
@@ -31,16 +32,25 @@ void init() {
...
+    Textures theme_texture = BackgroundTheme1 + level_theme;
+    int level_theme_position_ignore, level_theme_width, level_theme_height;
+    get_texture_dimensions(theme_texture, &level_theme_position_ignore,
+                           &level_theme_position_ignore, &level_theme_width,
+                           &level_theme_height);
+    for (int j = 0; j < win_surf->h; j += level_theme_height) {
+        for (int i = 0; i < win_surf->w; i += level_theme_width) {
+            draw_texture(win_surf, theme_texture, i, j, false);
         }
     }
```
Comment (maelehrhard) [inline arkanoid.c:55]: You should put this in a draw_background() method for more clarity 😎

File: arkanoid.c [hunk 3]
```diff
@@ -65,10 +75,25 @@ void draw() {
...
     draw_integer(win_surf, (int) get_current_fps(), 10 + fps_text_width, 40);
+
+    int offset_x = (win_surf->w - LEVEL_WIDTH * 32) / 2;
+    int offset_y = 150;
+    for (int y = level->offset; y < level->height + level->offset; y++) {
+        for (int x = 0; x < LEVEL_WIDTH; x++) {
+            if (level->bricks[y][x].type != EMPTY) {
+                int brick_x = offset_x + x * 32;
+                int brick_y = offset_y + y * 16;
+                draw_texture(win_surf, BallTexture, brick_x, brick_y, false);
```
Comment (maelehrhard) [inline arkanoid.c]: ? 😘

File: levels.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    brick.max_animation = 0;
+    brick.capsule_reward = capsule_reward;
+
+    if (type == METAL || type == GOLD) {
+        brick.durability = 8;
+        brick.max_animation = 6;
+    }
+
+    return brick;
+}
```
Comment (maelehrhard) [inline levels.c]: Why not this ?

```suggestion
    brick.type = type;
    brick.current_animation = 0;
    brick.capsule_reward = capsule_reward;

    if (type == METAL || type == GOLD) {
        brick.durability = 8;
        brick.max_animation = 6;
    } else {
        brick.durability = 1;
        brick.max_animation = 0;
    }
```

File: levels.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    // Theme
+    fgets(line, sizeof(line), file);
+    level.theme = atoi(line);
+
+    // Offset
+    fgets(line, sizeof(line), file);
+    level.offset = atoi(line);
+
+    // Height
```
Comment (maelehrhard) [inline levels.c:52]: I think you can remove those comments since we can read what the file line is about 🔥

**Review by maelehrhard — COMMENTED** [review]
🔥

File: levels.c (hunk 5 above)
Comment (axelmunch) [reply levels.c:52]: I'd rather not

Comment (maelehrhard) [reply levels.c:52]: 😒

**Review by maelehrhard — APPROVED** [review]

**Pull request merged**